### ISSUE TITLE
Fedora 36 - 37 not build

### ISSUE THREAD

**Issue by amidevous** [openssl/openssl#20047]
on Fedora 36 - 37 not build

```
wget -O openssl-3.0.7.tar.gz http://artfiles.org/openssl.org/source/openssl-3.0.7.tar.gz
tar -xvf openssl-3.0.7.tar.gz
cd /root/ffmpeg_sources/openssl-3.0.7
./Configure --prefix=/root/ffmpeg_build --openssldir=/root/ffmpeg_build/etc/pki/tls shared zlib
make -j$(nproc --all)
```

```
${LDCMD:-gcc} -pthread -m64 -Wa,--noexecstack -I/root/ffmpeg_build/include -L/root/ffmpeg_build/lib64  -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -Wl,-dT,/root/rpmbuild/BUILD/openssl-3.0.7/.package_note-xtream-ui-openssl3-3.0.7-1.fc36.x86_64.ld \
        -o test/moduleloadtest \
        test/moduleloadtest-bin-moduleloadtest.o \
        test/moduleloadtest-bin-simpledynamic.o \
        -lz -ldl -pthread
/usr/bin/ld: test/moduleloadtest-bin-moduleloadtest.o: relocation R_X86_64_32 against `.rodata' can not be used when making a PIE object; recompile with -fPIE
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:25101: test/moduleloadtest] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/root/rpmbuild/BUILD/openssl-3.0.7'
make: *** [Makefile:3220: build_sw] Error 2
```

please solve

**Comment by levitte** [comment]
I suspect you picked up something from the environment.  Please do the following and show us the output:

``` console
$ cd /root/ffmpeg_sources/openssl-3.0.7
$ perl configdata.pm --dump
```

**Comment by amidevous** [comment]
solve this work

fedora specific parameter use on rpm

```
%build
RPM_OPT_FLAGS="$RPM_OPT_FLAGS -Wa,--noexecstack -Wa,--generate-missing-build-notes=yes -DPURIFY $RPM_LD_FLAGS"
```

`./Configure etc $RPM_OPT_FLAGS`

this work on manual

`./Configure --prefix=/root/ffmpeg_build --openssldir=/root/ffmpeg_build/etc/pki/tls shared zlib -Wa,--noexecstack -Wa,--generate-missing-build-notes=yes -DPURIFY`

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
